apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -409,6 +416,20 @@ type FlavorFungibility struct {
 	// +kubebuilder:validation:Enum={Preempt,TryNextFlavor}
 	// +kubebuilder:default="TryNextFlavor"
 	WhenCanPreempt FlavorFungibilityPolicy `json:"whenCanPreempt,omitempty"`
+	// whenCanPreemptAndBorrow defines how should a workload chose the flavor in case of
+	// multiple options. If either borrowing or preemption is necessary and multiple options
+	// are available, then this field defines the selection strategy. Can be set only if
+	// `WhenCanBorrow = TryNextFlavor` and `WhenCanPreempt = TryNextFlavor`.
+	// The possible values are:
+	//
+	// - `PreferBorrowing`: prefer to allocate in a flavor that does not preempt
+	//    other workloads. This is the default flavor selection policy if
+	//    `WhenCanBorrow = TryNextFlavor` and `WhenCanPreempt = TryNextFlavor`.
+	// - `PreferPreemption`: prefer to allocate in a flavor that uses only the nominal quota
+	//    even if it means preempting other workloads.
+	// +optional
+	// +kubebuilder:validation:Enum={PreferBorrowing,PreferPreemption}
+	WhenCanPreemptAndBorrow FlavorSelectionPolicy `json:"whenCanPreemptAndBorrow,omitempty"`
```

Comment:
```suggestion
	WhenCanPreemptAndBorrow *FlavorSelectionPolicy `json:"whenCanPreemptAndBorrow,omitempty"`
```
It is generally good practice in k8s API to add new fields to pre-existing structures using pointer fields.
Otherwise the clients generated using old schema will be failing to get / patch. Also clients generated with the new code will have issues with old schema as they would always expect the field.